packages/eslint-plugin/src/rules/strict-boolean-expressions.ts

```diff
@@ -814,6 +853,14 @@ export default util.createRule<Options, MessageId>({
         }
       }
 
+      const enums = types.filter(type =>
+        tsutils.isTypeFlagSet(type, ts.TypeFlags.EnumLike),
+      );
+
+      if (enums.some(e => e)) {
```

Comment:
Thank you for your detailed explanation!
Since the change you suggested looks different from what you explained (because of using `.filter`),  I fixed the code as you explained below.
https://github.com/typescript-eslint/typescript-eslint/pull/6096/commits/be467bab791930839be36c0f24826f080332686b